File: C/MaximumDepthofBinaryTree.c
#include<stdio.h>
//definition for a binary tree
struct TreeNode
{
	int val;
	struct TreeNode *left;
	struct TreeNode *right;
};
int findDepth(struct TreeNode * root);
int main()
{
	int maxDepth(struct TreeNode * root);
}

int maxDepth(struct TreeNode * root)
{
	if(root!=NULL)root->val=1;
	else return 0;
	return findDepth(root);
}

int findDepth(struct TreeNode * root)
{
	static int depth=1;
	if(root->left!=NULL)
	{
		root->left->val=root->val+1;
		if(root->left->val>depth)depth=root->left->val;
		findDepth(root->left);
	}
	if(root->right!=NULL)
	{
		root->right->val=root->val+1;
		if(root->right->val>depth)depth=root->right->val;
		findDepth(root->right);
	}
	return depth;
}

File: Java/MaximumDepthofBinaryTree.java
//   Definition for a binary tree node
class TreeNode
{
	int val;
	TreeNode left;
	TreeNode right;
	TreeNode(int x){val = x;}
}

public class MaximumDepthofBinaryTree
{
	public int maxDepth(TreeNode root) {
		if(root == null)
			return 0;
		return 1 + max(maxDepth(root.left), maxDepth(root.right));
	}
	int max(int a,int b)
	{
		if(a>b)
			return a;
		else 
			return b;
	}
}
